internal/interceptors.go

```diff
@@ -97,8 +98,19 @@ type WorkflowOutboundCallsInterceptor interface {
 	GetLastError(ctx Context) error
 }
 
+type ActivityInterceptor interface {
+	InterceptActivity(ctx context.Context, next ActivityInboundCallsInterceptor) ActivityInboundCallsInterceptor
+}
+
+type ActivityInboundCallsInterceptor interface {
+	Init(ctx context.Context) error
+
+	ExecuteActivity(ctx context.Context, activityType string, args ...interface{}) []interface{}
```

Comment:
> In Java and Node this method is called `execute`

In Go, this is analogous to `ExecuteWorkflow`. Java has `WorkflowInboundCallsInterceptor.execute` and Go has `WorkflowInboundCallsInterceptor.ExecuteWorkflow` so for that same consistency, Java has `ActivityInboundCallsInterceptor.execute` and Go has `ActivityInboundCallsInterceptor.ExecuteActivity`.

> I'd also strongly consider using a struct for interceptor args

That would be inconsistent with the other interceptors e.g. `ExecuteWorkflow`. I intentionally sacrificed ergonomics for consistency.

Reply:
I strongly recommend we get this interceptor right now to avoid redoing this later, we'll need to change the other interceptors in go to resemble the Java / Node.js design (see #529).

Reply:
You haven't replied to my most important comment, the interceptor has to be able to prevent activity execution, probably by returning an error.

Reply:
Per off-PR discussion, looks like we're looking to land activity interceptors prior to redesigning the interceptor API.